src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardConsumer.java

```diff
@@ -133,6 +135,59 @@
                   Optional<Integer> maxGetRecordsThreadPool) {
         this.streamConfig = streamConfig;
         this.recordProcessor = recordProcessor;
+        this.recordsFetcherFactory = new SimpleRecordsFetcherFactory(streamConfig.getMaxRecords());
```

Comment:
Again creating multiple instances of the factory beats the purpose of creating the factory in the first place. Try and pass the client lib config around.

Reply:
Same reason here.